addon/macros/feature-flag.js

```diff
@@ -0,0 +1,21 @@
+import { camelize } from '@ember/string';
+import { computed, get, set } from '@ember/object';
+import { getOwner } from '@ember/application';
+
+const featuresServiceProperty = `_features_${+(new Date())}`
+
+export default function featureFlagMacro(flag) {
+  const flagPath = `${featuresServiceProperty}.${camelize(flag)}`;
+
+  return computed(flagPath, function() {
+    ensureFeaturesInjection(this);
+    return get(this, flagPath);
+  });
+}
+
+function ensureFeaturesInjection(object) {
+  if (!get(object, featuresServiceProperty)) {
+    const features = getOwner(object).lookup('service:features')
+    set(object, featuresServiceProperty, features)
+  }
+}
```

Comment:
I also considered an implementation more like `helper:feature-flags`, where we directly manage observers. The challenge is calling `removeObserver` during destruction. The best I came up with was

```js
const now = +(new Date())
const dependencyName = `_feature-flag-macro-dependency-${now}`
const callbackName = `_feature-flag-macro-callback-${now}`

function callback = function() {
  this.notifyPropertyChange(dependencyName);
}

export default function featureFlagMacro(flag) {
  const normalizedFlag = camelize(flag);
  let observerAdded = false;
  let originalWillDestroy = null;

  return computed(dependencyName, function(propertyName) {
    const features = getOwner(this).lookup('service:features');

    if (!this[callbackName]) {
      this[callbackName] = callback;
    }

    if (!observerAdded) {
      observer = features.addObserver(normalizedFlag, this, callbackName)
      observerAdded = true
    }

    if (!originalWillDestroy) {
      originalWillDestroy = this.willDestroy;
      this.willDestroy = function() {
        features.removeObserver(normalizedFlag, this, callbackName);
        return originalWillDestroy.call(this, ...arguments);
      };
    }
  });
}
```
It seems more complicated, but I'm happy to switch to that if you prefer.